doc/developer-documentation.md

```diff
@@ -50,3 +50,72 @@ build_configs:
 
 That's it! The tree is enabled now.
 All the jobs defined under `jobs` section of [config file](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) would run on the newly added tree until specified otherwise.
+
+
+## Enabling a new test
+
+KernelCI currently runs several test suites. This manual is intended
+to provide documentation for developers on how to enable new tests.
+
+
+### Job definition
+The pipeline [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file has `jobs` section for defining various build and test jobs. The currently enabled tests are `kver`, `baseline`, `kunit`, `kselftest`, and `sleep` tests.
+`ChromeOS` specific job definitions are located in [config/jobs-chromeos.yaml](https://github.com/kernelci/kernelci-pipeline/blob/main/config/jobs-chromeos.yaml) file.
+
+To enable a new test, an entry needs to be added as follows:
+
+```yaml
+jobs:
+  <job-name>:
+    template: <job-template>
+    kind: <kind of job, either 'kbuild', 'job' or 'test'>
+    image: <docker-image-name>
+    params: <parameters such as arch, compiler, defconfig>
+    rules: <job-rules>
+    kcidb_test_suite: <kcidb-mapping>
+```
+
+Here is the description of each field:
+- A `jinja2` template should be added to [`config/runtime`](https://github.com/kernelci/kernelci-pipeline/tree/main/config/runtime) directory.
+The template will be used to generate test definition.
+- `kind` field denotes type of the job. It should be `kbuild` for build jobs, `job` for a test suite, and simply `test` for a single test case.
+- `image` field specifies the Docker image used for building and running the test.
```

Comment:
It might be useful to mark this field as optional - LAVA TestJobs wouldn't use this field but the one from their definition.

Reply:
That's true. I've updated it.